test-fixtures/src/bin/geolocation-lookup.rs

```diff
@@ -0,0 +1,40 @@
+//! A guest program to test that Geolocation lookups work properly.
+
+use fastly::geo::{
+    geo_lookup,
+    ConnSpeed,
+    ConnType,
+    Continent,
+    ProxyDescription,
+    ProxyType,
+    UtcOffset,
+};
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+fn main() {
+    let client_ip_v4 = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
+    let geo_v4 = geo_lookup(client_ip_v4).unwrap();
+    assert_eq!(geo_v4.as_name(), "Fastly Test");
+    assert_eq!(geo_v4.as_number(), 12345);
+    assert_eq!(geo_v4.area_code(), 123);
+    assert_eq!(geo_v4.city(), "Test City");
+    assert_eq!(geo_v4.conn_speed(), ConnSpeed::Broadband);
+    assert_eq!(geo_v4.conn_type(), ConnType::Wired);
+    assert_eq!(geo_v4.continent(), Continent::NorthAmerica);
+    assert_eq!(geo_v4.country_code(), "CA");
+    assert_eq!(geo_v4.country_code3(), "CAN");
+    assert_eq!(geo_v4.country_name(), "Canada");
+    assert_eq!(geo_v4.latitude(), 12.345);
+    assert_eq!(geo_v4.longitude(), 54.321);
+    assert_eq!(geo_v4.metro_code(), 0);
+    assert_eq!(geo_v4.postal_code(), "12345");
+    assert_eq!(geo_v4.proxy_description(), ProxyDescription::Unknown);
+    assert_eq!(geo_v4.proxy_type(), ProxyType::Unknown);
+    assert_eq!(geo_v4.region(), Some("CA-BC"));
+    // assert_eq!(geo_v4.utc_offset(), Some(UtcOffset::from_hms(-7, 0, 0).unwrap()));
```

Comment:
@mgattozzi For some reason this line fails. I even tried an IP address in California with Compute@Edge and the result was wrong. Perhaps an issue with Rust SDK?

Reply:
I saw this before and I tried digging into it a little bit (not too much mind you), but I'm not sure whether it's a bug in the SDK (I haven't tried our JS one to check) or if it is totally correct. If you're seeing this on C@E then I think we can just leave this as is for now and I can open up an issue on our internal tracker to investigate it more. You can put a small note saying why it's commented out.